File: Windows Projects/Windows Sample Projects/Pass non-string parameters between pages in Windows Phone 8 app/C#/CSWP8PassNonstringpara/NavigationExtensions.cs
/****************************** Module Header ******************************\
* Module Name:    NavigationExtensions.cs
* Project:        CSWP8PassNonstringpara
* Copyright (c) Microsoft Corporation
*
* This sample will demo how to pass no-string data between two pages.
* This is custom extension for NavigationService.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\*****************************************************************************/

using System;
using System.Windows.Navigation;

namespace CSWP8PassNonstringpara
{
    public static class NavigationExtensions
    {
        // Store parameters to be passed.
        private static object _navigationData = null;

        /// <summary>
        /// Set data.
        /// </summary>
        /// <param name="service">NavigationService</param>
        /// <param name="page">Target page.</param>
        /// <param name="data">Parameter data.</param>
        public static void Navigate(this NavigationService service, string page, object data)
        {
            _navigationData = data;
            try
            {
                service.Navigate(new Uri(page, UriKind.Relative));
            }
            catch (Exception)
            {
                throw;
            }
        }

        /// <summary>
        /// Get data.
        /// </summary>
        /// <param name="service">NavigationService</param>
        /// <returns></returns>
        public static object GetLastNavigationData(this NavigationService service)
        {
            object data = _navigationData;
            // Reset
            _navigationData = null;
            return data;
        }
    }

}

File: Windows Projects/Windows Sample Projects/Pass non-string parameters between pages in Windows Phone 8 app-2/VB/VBWP8PassNonstringpara/NavigationExtensions.vb
'****************************** Module Header ******************************\
' Module Name:    NavigationExtensions.vb
' Project:        VBWP8PassNonstringpara
' Copyright (c) Microsoft Corporation
'
' This sample will demo how to pass no-string data between two pages.
' This is custom extension for NavigationService.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'*****************************************************************************/

Public Module NavigationExtensions
    ' Store parameters to be passed.
    Private _navigationData As Object = Nothing

    ''' <summary>
    ''' Set data.
    ''' </summary>
    ''' <param name="service">NavigationService</param>
    ''' <param name="page">Target page.</param>
    ''' <param name="data">Parameter data.</param>
    <System.Runtime.CompilerServices.Extension> _
    Public Sub Navigate(service As NavigationService, page As String, data As Object)
        _navigationData = data
        Try
            service.Navigate(New Uri(page, UriKind.Relative))
        Catch generatedExceptionName As Exception
            Throw
        End Try
    End Sub

    ''' <summary>
    ''' Get data.
    ''' </summary>
    ''' <param name="service">NavigationService</param>
    ''' <returns></returns>
    <System.Runtime.CompilerServices.Extension> _
    Public Function GetLastNavigationData(service As NavigationService) As Object
        Dim data As Object = _navigationData
        ' Reset
        _navigationData = Nothing
        Return data
    End Function
End Module


